apiserver/facades/agent/caasagent/caasagent.go

```diff
@@ -4,14 +4,173 @@
 package caasagent
 
 import (
+	"context"
+
+	"github.com/juju/names/v6"
+
 	"github.com/juju/juju/apiserver/common"
-	"github.com/juju/juju/apiserver/common/cloudspec"
-	"github.com/juju/juju/apiserver/common/model"
+	commonmodel "github.com/juju/juju/apiserver/common/model"
+	apiservererrors "github.com/juju/juju/apiserver/errors"
+	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/apiserver/internal"
+	"github.com/juju/juju/controller"
+	"github.com/juju/juju/core/crossmodel"
+	coremodel "github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/network"
+	corewatcher "github.com/juju/juju/core/watcher"
+	environscloudspec "github.com/juju/juju/environs/cloudspec"
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/rpc/params"
+	"github.com/juju/juju/state"
 )
 
-// FacadeV2 is the V2 facade of the caas agent
+// StubService will be replaced once the implementation is finished.
+type StubService interface {
+	// CloudSpec returns the cloud spec for the model.
+	CloudSpec(ctx context.Context) (environscloudspec.CloudSpec, error)
+}
+
+// ModelService represents the credential service provided by the
+// provider.
+type ModelService interface {
+	WatchModelCloudCredential(ctx context.Context, modelUUID coremodel.UUID) (corewatcher.NotifyWatcher, error)
```

Comment:
This may not be an ideal request. But I want to see this wrapped into the API to remove the model uuid param. This facade is scoped and the params structure we accept for watching model cloud credentials takes a model uuid for some unknown reason.

This all scares me that mistakes can be introduced. Is it possible to wrap this up inside of register with the model uuid of context so that inside the facade you don't have the ability to step side ways?

Reply:
Would also love a comment.